src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -106,6 +108,112 @@ public class RuntimeConfigLoaderJsonDeserializerTests
+        [DataRow(false, "dwsql", DisplayName = "Do not replace environment variables when datasource option is not given.")]
+        [DataRow(true, "dwsql", DisplayName = "Replace environment variables when datasource option is not given.")]
+        public void TestConfigParsingWithEnvVarReplacement(bool replaceEnvVar, string databaseType)
+        {
+            // Act
+            SetEnvironmentVariablesFromDictionary(_environmentFileContentDict);
+            string configWithEnvVar = _configWithVariableDataSource.Replace("{0}", GetDataSourceConfigForGivenDatabase(databaseType));
+
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(
+              configWithEnvVar, out RuntimeConfig runtimeConfig, replaceEnvVar: replaceEnvVar));
+
+            // Assert
+            switch (databaseType)
+            {
+                case "mssql":
+                    Assert.AreEqual(runtimeConfig.DataSource.DatabaseType, DatabaseType.MSSQL);
+                    Assert.AreEqual(runtimeConfig.DataSource.Options["set-session-context"].ToString().ToLower(), GetExpectedPropertyValue("MSSQL_SET_SESSION_CONTEXT", replaceEnvVar).ToLower());
+                    break;
+                case "cosmosdb_nosql":
+                    Assert.AreEqual(runtimeConfig.DataSource.DatabaseType, DatabaseType.CosmosDB_NoSQL);
+                    Assert.AreEqual(runtimeConfig.DataSource.Options["database"].ToString(), GetExpectedPropertyValue("DATABASE_NAME", replaceEnvVar));
+                    Assert.AreEqual(runtimeConfig.DataSource.Options["container"].ToString(), GetExpectedPropertyValue("DATABASE_CONTAINER", replaceEnvVar));
+                    Assert.AreEqual(runtimeConfig.DataSource.Options["schema"].ToString(), GetExpectedPropertyValue("GRAPHQL_SCHEMA_PATH", replaceEnvVar));
+                    break;
+                case "mysql":
+                    Assert.AreEqual(runtimeConfig.DataSource.DatabaseType, DatabaseType.MySQL);
+                    Assert.AreEqual(runtimeConfig.DataSource.Options.Count, 0);
+                    break;
+                case "postgresql":
+                    Assert.AreEqual(runtimeConfig.DataSource.DatabaseType, DatabaseType.PostgreSQL);
+                    Assert.AreEqual(runtimeConfig.DataSource.Options, null);
+                    break;
+                case "dwsql":
+                    Assert.AreEqual(runtimeConfig.DataSource.DatabaseType, DatabaseType.DWSQL);
+                    Assert.AreEqual(runtimeConfig.DataSource.Options, null);
+                    break;
+            }
+
+            // Cleanup
+            ClearEnvironmentVariablesFromDictionary(_environmentFileContentDict);
+        }
+
+        /// <summary>
+        /// Test the parsing of DataSource section in runtime config for cosmosdb_nosql
+        /// where under cosmosDb options has database as null, container is not provided, and schema is an empty string.
```

Comment:
What do you expect the behavior to be when such a config is provided?The function name states `DataSourceOptionsForCosmosDBContainsInvalidValues` can you elaborate on:
- why no error is expected
- how is dab handling 'invalid values' for options?